core/domain/role_services.py

```diff
@@ -131,3 +142,69 @@ def get_all_actions(role):
         role_actions.extend(get_all_actions(parent_role))
 
     return list(set(role_actions))
+
+
+def get_updatable_roles():
+    """Returns roles to which a user's role can be updated by admin interface.
+
+    Returns:
+        dict. A dict of all roles that a user can be given by admin.
+            key -> roles in format as stored in backed.
+            value -> roles in format to show users.
+    """
+    updatable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_GUEST]:
+            updatable_roles[role] = get_human_readable_role(role)
+    return updatable_roles
+
+
+def get_viewable_roles():
+    """Returns roles by which users can be searched in admin interface.
```

Comment:
Try to explain what this function returns in isolation, rather than where it is used. All you care about here is that this function returns a dict mapping role IDs to human-readable names. It could take an additional argument listing the roles it should include. This will keep it self-contained and not dependent on some kind of global context.